src/ReflectionFunction.php

```diff
@@ -18,25 +18,46 @@
 /**
  * AST-based reflection for function
  */
-class ReflectionFunction extends BaseReflectionFunction
+class ReflectionFunction extends BaseReflectionFunction implements IReflection
 {
     use ReflectionFunctionLikeTrait, InternalPropertiesEmulationTrait;
 
     /**
+     * Name of the function
+     *
+     * @var string|\Closure
+     */
+    private $functionName;
+
+    /**
      * Initializes reflection instance for given AST-node
      *
      * @param string|\Closure $functionName The name of the function to reflect or a closure.
      * @param Function_|null  $functionNode Function node AST
      */
-    public function __construct($functionName, Function_ $functionNode)
+    public function __construct($functionName, Function_ $functionNode = null)
     {
-        $namespaceParts = explode('\\', $functionName);
+        $this->functionName = $functionName;
+        $namespaceParts     = explode('\\', $functionName);
         // Remove the last one part with function name
         array_pop($namespaceParts);
         $this->namespaceName = join('\\', $namespaceParts);
-
         $this->functionLikeNode = $functionNode;
+        if (!$this->functionLikeNode) {
+            $isUserDefined = true;
+            if ($this->wasIncluded()) {
+                $nativeRef = new BaseReflectionFunction($functionName);
+                $isUserDefined = $nativeRef->isUserDefined();
+            }
+            if ($isUserDefined) {
```

Comment:
How is `user defined` correlate with `!$this->functionLikeNode`?

Reply:
For a fully constructed `ReflectionFunction` object,  `$this->isUserDefined()` should always return `!$this->functionLikeNode`.  I believe I am actually returning the equivalent of `!($this->functionLikeNode || !parent::isUserDefined())` just in case there some condition I didn't account for, but too my knowledge these should always be the same.